client/client.go

```diff
@@ -423,13 +425,83 @@ func (c *Client) getLocalMeta() error {
 			c.loadTargets(targets.Targets)
 		}
 	}
+
+	if loadFailed {
+		// If any of the metadata failed to be verified, return the reason for that failure
+		// and fail fast before delegated targets
+		return retErr
+	}
+
+	// verifiedDelegatedTargets is a set of verified delegated targets
+	verifiedDelegatedTargets := make(map[string]bool)
+	for fileName := range meta {
+		if roles.IsDelegatedTargetsManifest(fileName) {
+			if delegationPath, err := c.getDelegationPathFromRaw(snapshot, meta[fileName]); err != nil {
```

Comment:
I think, as we discussed earlier, if we note in an issue that we don't record delegated targets roles that don't ultimately lead to any actual targets, then we should be good. Thanks!

Reply:
+1 

I had to convince myself about this: I realize it probably isn't too big of a deal. If there's no ultimate target in that delegation path, then there really was no need to ever store them locally for verifying a target.

Reply:
@BaptisteFoy can you pls create a new issue for this? thx!

Reply:
Created https://github.com/theupdateframework/go-tuf/issues/444 to track this

Reply:
thanks!